Sources/Alchemy/Rune/Model/Model+CRUD.swift

```diff
@@ -18,15 +18,16 @@ extension Model {
         VALUES (\(fields.enumerated().map { index, _ in "$\(index + 1)" }.joined(separator: ", ")))
         """
 
-        return db.query(statement, values: fields.map { $0.value }, on: loop)
+        return db.runQuery(statement, values: fields.map { $0.value }, on: loop)
             .voided()
     }
     
     /// Deletes this model from the database.
-    func delete(db: Database = DB.default, loop: EventLoop = Loop.current) -> EventLoopFuture<Void> {
+    func delete(db: Database = DB.default, loop: EventLoop = Loop.current) throws -> EventLoopFuture<Void> {
```

Comment:
I think we should avoid functions throwing & returning an ELF; it's two sources of errors that the user has to handle. Ideally this would collapse the error into the ELF. Ugly on the inside but easier to digest for the consumer (errors would only come from one place, the ELF).

Something like...
```swift
func delete(db: Database = DB.default, loop: EventLoop = Loop.current) -> EventLoopFuture<Void> {
        do {
            let id = try self.idField()
            return try Self.query(database: db)
                .where(idField.column == idField.date())
                .delete(on: loop)
                        .voided()
        } catch {
            return self.eventLoop.future(error)
        }
    }
```
Can add some `EventLoop` sugar like...
```swift
extension EventLoop {
    /// Takes a throwing closure and turns that into an event loop future; a succeeded one if the closure
    /// didn't throw, or an errored one if the closure did.
    func future<T>(_ throwingClosure: () throws -> T ) -> EventLoopFuture<T> {
        do {
            return self.future(try throwingClosure())
        }
        catch {
            return self.future(error: error)
        }
    }
}
```
so this function would then be...
```swift
func delete(db: Database = DB.default, loop: EventLoop = Loop.current) throws -> EventLoopFuture<Void> {
        loop.future(try self.idField)
            .flatMap {
                /// Assuming the same thing was applied to `query` (piping error to ELF instead of throws in signature)
                Self.query(database: db)
                    .where(idField.column == idField.date())
                    .delete(on: loop)
                    .voided()
            }
    }
```

Probably a better form of sugar for this (if only swift had decorators...), but you get the point.

Reply:
If you look at Insert the direction I sent was the first option you provided. Looks like last night I just forgot to carry it all the way through. Was using failed futures basically